packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -529,26 +525,107 @@ class RenderListWheelViewport
     size = constraints.biggest;
   }
 
+  /// Get the index of a child by looking at its parentData.
+  int indexOf(RenderBox child) {
+    assert(child != null);
+    final ListWheelParentData childParentData = child.parentData;
+    assert(childParentData.index != null);
+    return childParentData.index;
+  }
+
+  /// Return the index of the child that should be at the given offset.
+  int offsetToIndex(double offset) => offset ~/ itemExtent;
+
+  /// Return the vertical offset of the child with the given index.
+  double indexToOffset(int index) => index * itemExtent;
+
+  void _createChild(int index, {RenderBox after}) {
+    invokeLayoutCallback<BoxConstraints>((BoxConstraints constraints) {
+      assert(constraints == this.constraints);
+      childManager.createChild(index, after: after);
+    });
+  }
+
+  void _destroyChild(RenderBox child) {
+    invokeLayoutCallback<BoxConstraints>((BoxConstraints constraints) {
+      assert(constraints == this.constraints);
+      childManager.removeChild(child);
+    });
+  }
+
+  /// Layout base on children provided by [childManager].
+  ///
+  /// From the current scroll offset, we can calculate the minimum index and
+  /// maximum index that is visible in the viewport. We can also get the index
+  /// range of the children that are currently active.
+  /// We will modify the child list so that children that are not visible
+  /// will be removed, and those that should visible but not yet created will be
+  /// instantiate by the childManager.
   @override
   void performLayout() {
-    double currentOffset = 0.0;
-    RenderBox child = firstChild;
     final BoxConstraints innerConstraints =
-        constraints.copyWith(
-          minHeight: _itemExtent,
-          maxHeight: _itemExtent,
-          minWidth: 0.0,
-        );
-    while (child != null) {
+    constraints.copyWith(
+      minHeight: _itemExtent,
+      maxHeight: _itemExtent,
+      minWidth: 0.0,
+    );
+
+    int targetFirstIndex = offsetToIndex(offset.pixels - size.height / 2 - _itemExtent / 2);
+    int targetLastIndex = offsetToIndex(offset.pixels + size.height / 2 - _itemExtent / 2);
+
+    if(childManager.childCount != null) {
+      targetFirstIndex = targetFirstIndex.clamp(0, childManager.childCount - 1);
+      targetLastIndex = targetLastIndex.clamp(0, childManager.childCount - 1);
+    }
+
+    if(firstChild == null) {
```

Comment:
Sprinkle code comments liberally. Document that this is for when it's laying out for the first time and no render object children are 'cached' yet or some such.

Reply:
done.